functions/minsurf_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package functions
+
+import (
+	"math"
+	"testing"
+
+	"github.com/gonum/diff/fd"
+	"github.com/gonum/floats"
+)
+
+func TestMinimalSurface(t *testing.T) {
```

Comment:
Strictly speaking, I shouldn't be testing if the gradient at ExactX() is small (especially at such coarse grid resolutions), because ExactX is the exact solution projected on the grid, not the exact solution to the discrete problem. I will reconsider it after the weekend.